src/System.Collections/tests/Generic/HashSet/HashSet.Generic.Tests.netcoreapp.cs

```diff
@@ -97,7 +97,7 @@ public void HashSet_Generic_TryGetValue_Contains()
             Assert.Equal(value, actualValue);
             if (!typeof(T).IsValueType)
             {
-                Assert.Same(value, actualValue);
```

Comment:
I see your PR description, but I don't understand this change.  Can you please elaborate on why you changed these to Equal?  If it's really the case that there's no benefit to Same, then the whole if block should be deleted because we already do this exact Equal comparison two lines above.  But it does seem like there's a benefit to using Same here when T is a reference type.